spec/models/course/forum/topic_spec.rb

```diff
@@ -37,6 +37,18 @@
         end
       end
 
+      context 'when reinitializing a topic' do
+        subject do
+          test_topic = Course::Forum::Topic.new(posts_attributes: [title: nil])
+          test_topic.save
```

Comment:
I did this because when use nested attributes and fail to save the records, the `after_initialize` call back will be triggered again.
These actions imitate the same things I did in the topic controller.